common/config/src/argv.rs

```diff
@@ -207,6 +208,11 @@ pub struct ArgumentOptions {
     /// Defaults to /tmp/logdna.
     #[structopt(long, env = env::RETRY_DIR)]
     retry_dir: Option<String>,
+
+    /// When set, limits the amount of disk space the agent will use to store log lines that
+    /// need to be resent to the ingestion API.
+    #[structopt(long, env = env::RETRY_DISK_LIMIT)]
+    retry_disk_limit: Option<u64>,
```

Comment:
What do you think about naming it with the unit?
`retry_disk_limit_mb` for example

Reply:
Should we default to a super super large value, like 2 or 4GiB.
I think having unlimited storage usage by default is unexpected and during long outages can lead to infra issues on the user side.

Reply:
> Should we default to a super super large value, like 2 or 4GiB.
> I think having unlimited storage usage by default is unexpected and during long outages can lead to infra issues on the user side.

We discussed this at standup today and agree that filling the disk space up is problematic but that enforcing a default might lead to data loss if we pick incorrectly. Customers who are sensitive to their disk usage can monitor the disk used as well as set limits for their configurations. Tom felt that running the risk of consuming the disk is less bad than failing to ingest logs.

Reply:
> What do you think about naming it with the unit?

@teeohhem Commit https://github.com/logdna/logdna-agent-v2/commit/8965a89b0ecfa3ac842b3db4e1d5eb5426b0d4f6 is adding support for using humanize. Does that address your concerns?